tests/sources/xml/test_marc.py

```diff
@@ -1010,6 +1010,112 @@ def test_get_contents_transforms_correctly_if_fields_missing():
+                <subfield code="e">composer,</subfield>
+                <subfield code="e">performer.</subfield>
+            </datafield>
+            <datafield tag="110" ind1="2" ind2=" ">
+                <subfield code="a">Ceeys (Musical group),</subfield>
+                <subfield code="e">composer,</subfield>
+                <subfield code="e">performer.</subfield>
+            </datafield>
+            <datafield tag="111" ind1="2" ind2=" ">
+                <subfield code="a">Theory of Cryptography Conference</subfield>
+                <subfield code="n">(5th :</subfield>
+                <subfield code="d">2008 :</subfield>
+                <subfield code="c">New York, N.Y.)</subfield>
+            </datafield>
+            <datafield tag="700" ind1="1" ind2=" ">
+                <subfield code="a">Binelli, Daniel,</subfield>
+                <subfield code="e">arranger of music,</subfield>
+                <subfield code="e">composer,</subfield>
+                <subfield code="e">instrumentalist.</subfield>
+            </datafield>
+            <datafield tag="711" ind1="2" ind2=" ">
+                <subfield code="a">Bayreuther Festspiele.</subfield>
+                <subfield code="e">Orchester,</subfield>
+                <subfield code="e">instrumentalist.</subfield>
+            </datafield>
+            """
+        )
+    )
+    assert Marc.get_contributors(source_record) == [
+        timdex.Contributor(value="Tran, Phong", kind="composer"),
+        timdex.Contributor(value="Tran, Phong", kind="performer"),
+        timdex.Contributor(value="Ceeys (Musical group)", kind="composer"),
+        timdex.Contributor(value="Ceeys (Musical group)", kind="performer"),
+        timdex.Contributor(
+            value="Theory of Cryptography Conference 2008 : New York, N.Y.)",
+            kind="Not specified",
+        ),
+        timdex.Contributor(value="Binelli, Daniel", kind="arranger of music"),
+        timdex.Contributor(value="Binelli, Daniel", kind="composer"),
+        timdex.Contributor(value="Binelli, Daniel", kind="instrumentalist"),
+        timdex.Contributor(value="Bayreuther Festspiele", kind="instrumentalist"),
+        timdex.Contributor(value="Bayreuther Festspiele", kind="Orchester"),
+    ]
+
+
+def test_get_contributors_transforms_correctly_if_fields_blank():
+    source_record = create_marc_source_record_stub(
+        datafield_insert=(
+            """
+            <datafield tag="100" ind1="1" ind2=" ">
+                <subfield code="a"></subfield>
+                <subfield code="e">composer,</subfield>
+            </datafield>
+            """
+        )
+    )
+    assert Marc.get_contributors(source_record) is None
+
+
+def test_get_contributors_transforms_correctly_if_fields_missing():
+    source_record = create_marc_source_record_stub()
+    assert Marc.get_contributors(source_record) is None
+
+
+def test_get_contributors_transforms_correctly_if_contributor_in_multiple_fields():
```

Comment:
I might rename to `..._if_contributor_multiple_roles` or `..._if_contributor_multiple_kinds`

Reply:
I like it! Renamed the test to `..._if_contributor_multiple_kinds`; "kinds" feels more explicit.